### DISCUSSION TITLE
Solaras visual not working

### DISCUSSION THREAD

**Discussion by DocZubes** [mesa/mesa#2484] [Q&A]
Hello, I need quick help, want to show my class the MESA program tomorrow and I saw that there was a massive update on saturday, really great stuff only one problem. THe Solara or more especially the SolaraViz isn't working. I wanted to give and show my students some of the examples, the wolf_sheep and boid_flockers since the plan was to later create a ContinuousSpace forest fire and agents to combat it, I saw that you now add solara to make it possible to use ContinuousSpace since previous I had to make my own. any ways, neither example works and then I decided to try any of the examples and  most use the Solara  run app.py and it seems like its not working. the terminal is stating ''Could not import SolaraViz. If you need it, install with 'pip install --pre mesa[viz]''' but I can confirm I have everything installed. 
![Screenshot 2024-11-11 114340](https://github.com/user-attachments/assets/b9b4a35e-1ee0-470b-b9b7-1bf7cf7a4b69)

I would really like to use this for the classroom but if I can't have this fix then we got to go back to Netlogo only. I am hoping its something on my end or maybe its already been pulled since writing this message.

**Comment by EwoutH** [comment]
> the terminal is stating ''Could not import SolaraViz. If you need it, install with 'pip install --pre mesa[viz]''' but I can confirm I have everything installed.

Could you show a list of installed packages or imports? `solara` should be in it.

If you're 100% sure everything is installed, could you give us a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example)?

**Edit:** You're teaching a class? In that case, do you want to set up a video call to solve this?

**Reply by EwoutH** [reply]
@DocZubes First things first: Which Mesa version are you using? Could you install our latest, 3.0.1?

**Reply by DocZubes** [reply]
Sorry its two screenshots VS doesn't let me zone out to show all installs but they are there

MESA
![image](https://github.com/user-attachments/assets/a735d0e2-5fd8-4bdb-88e5-a506f370cdbf)

Solara
![image](https://github.com/user-attachments/assets/6832d2b3-8941-4b73-b870-1b6173b60f91)


A video call sounds perfect, I can also make a example to test too.

**Reply by EwoutH** [reply]
Send me a message on Element, we can call right now: https://app.element.io/#/room/#project-mesa:matrix.org

**Reply by DocZubes** [reply]
before I forget, here is the full error for the example boid_flockers:

![image](https://github.com/user-attachments/assets/11891ada-b875-4c2a-bb90-6e238b679776)
![image](https://github.com/user-attachments/assets/febc956e-a68d-4951-ac0e-8e63327554eb)
![image](https://github.com/user-attachments/assets/734115e4-9f3d-47ec-b2cd-6b8ee36524d8)

**Reply by Corvince** [reply]
In the last line you can see you are missing networkx. You shouldn't need it and we need to fix this in mesa, but for now can you just install networkx and report if that resolves your issues?

**Reply by EwoutH** [reply]
@Corvince we did a call, it's more complicated than that. Multiple things are broken.

I do suspect a path or environment issue could be the cause, on my end I couldn't reproduce it.

**Reply by DocZubes** [reply]
So funny enough I was about to say I now installed it and tested to see if that was the issuse and it do open solaras now but now there is an error with the components, specifically, it is being given 2 positional arguments, while it only expects 1.

![image](https://github.com/user-attachments/assets/efce83f6-48f9-41b1-a66a-dadfcedeba5e)

Solaras' localhost:
![image](https://github.com/user-attachments/assets/d700f24e-4504-4aac-a42f-29e4c8466028)

**Reply by EwoutH** [reply]
That's the same error I had in 3.0.1 and have fixed in https://github.com/projectmesa/mesa/pull/2485.

However, if you fix it there's another error with 3.0.1.

Could you install 3.0.0 in the same way and see if it works?

**Reply by DocZubes** [reply]
Ok so I just had to updated mesa examples files and installed 3.0.0, it works now where solara does open and it can run but there is a problem where the visual is missing for the boid_flockers example on the localhost, does simulation just no visual:

![image](https://github.com/user-attachments/assets/955b947a-01af-4082-8096-46ae5e06040a)


I shall try virus on network since I see that was  updated also

update: virus on network has the same visual problem, it runs the simulation like a charm: 

![image](https://github.com/user-attachments/assets/30af2f07-9506-4027-ab21-f6db67844994)

**Reply by EwoutH** [reply]
You're discovering so much in a single day we didn't encounter in weeks of pre-release testing.

Which browser are you using?

**Reply by DocZubes** [reply]
I am am just using google chrome

**Reply by EwoutH** [reply]
I fixed the bug in boid_flockers, on my end it's now working with both 3.0.1: https://github.com/projectmesa/mesa/pull/2492

I have no idea what's causing this. @Corvince, @tpike3 or @quaquel ever seen this before?

**Reply by quaquel** [reply]
No never seen this before. It seems everything is rendered correctly. My hunch is some kind of add-blocking plugin is preventing images from displaying because the png's the matplotlib components are not showing. 

I would try the URL in a different browser with all plugins disabled and see what happens

**Reply by EwoutH** [reply]
If you can get it working, boid flockers should work if you install the latest mesa from our maintenance branch:

```
pip install -U -e git+https://github.com/projectmesa/mesa@3.0.x-maintenance#egg=mesa
```

**Reply by DocZubes** [reply]
Ok sorry was test a lot. So I opened the url to other places and amde sure all my add-blocking plugin, only one was on Opera GX, but on all them the visuals are missing, also on https://github.com/projectmesa/mesa/pull/2492#issue-2649371821, I ran it on 3.0.1 and it has this problem: 

![image](https://github.com/user-attachments/assets/fc9eeaa2-e873-471b-a70f-ace4d2710629)
 
So it seems boids still don't have neighbors, but I still tested 3.0.1 on the  other examples to see if visuals work on them

On schelling:

![image](https://github.com/user-attachments/assets/020b4a02-e791-4e39-a312-ea1e606eb69d)

On conways_game_of_life

![image](https://github.com/user-attachments/assets/c7d01c7e-06c8-4959-938b-b12e39ca69e5)


Both work just still no visual to see, I will test this on my laptop to see if its just my computer.

**Reply by EwoutH** [reply]
> I ran it on 3.0.1 and it has this problem:

 I fixed that with #2492, which you can install with: `pip install -U -e git+https://github.com/projectmesa/mesa@3.0.x-maintenance#egg=mesa`

I'm also releasing 3.0.2 in a bit.

<hr>

I really don't know what's happening here and why it isn't showing. Please open a new issue, and include as much information for us to make it reproducible.

One thing you can still try is running in a Jupyter Notebook, like the [Visualization Tutorial](https://mesa.readthedocs.io/latest/tutorials/visualization_tutorial.html) does. Maybe that works?

**Reply by DocZubes** [reply]
So I decide to be simple and reinstall all the packages that are needed for mesa, and it worked now it shows on my computer and I tested also on my laptop, so thanks for all the help, just one quick question, is there a way to increase framerate during the simulation, I remember there was away before but maybe there's a new way now
![image](https://github.com/user-attachments/assets/f966f1b1-6fe5-4632-b4de-341d243d6e03)

**Reply by quaquel** [reply]
Excellent!

and yes, via `play_interval` which is a keyword argument on `SolaraViz`

**Reply by EwoutH** [reply]
Glad to hear it’s now working!

Maybe we can add it as a slider, I also noticed it was slow. Together with seed, to maybe all models.

**Reply by EwoutH** [reply]
Did you get everything working you wanted? Any insights/feedback?

**Reply by DocZubes** [reply]
Yes!! I was able to show them all the examples so that the students can decide what type of forest fire simulation they would like proform in the following term. Thank you again for the quick help and fixes. I have no feedback, I know one group really wants to explore the visualization aspect with other packages/ create their own, so I might push them this way in the future if they have questions.

**Reply by EwoutH** [reply]
Great to hear! We're very open to both ideas and pull request on how we can improve it.

If any group develops a great example, a PR to [mesa-examples](https://github.com/projectmesa/mesa-examples) is also very welcome.

**Reply by RamGroningen** [reply]
Hello, I'm seeing the same issue currently with Mesa where Solara opens and shows the sidebar panel but no simulation is displayed. I tried in multiple browsers (chrome and firefox) and tried disabling adblock but no difference. I also unisntalled solara and mesa and install both manually but no luck. 

I have mesa version 3.5.1 installed and Solara version 1.61.0. Let me know if you have an idea of how to fix

I'm running the example app.py code from the game of life example as it is on the docs. I tried the same with the wealth model and both showed blank simulation screens with the sidebar displayed

**Reply by RamGroningen** [reply]
Ah looks like a dependency issue that will be fixed when the version upgrade is finished but for now can reinstall the pip ipyvue library as suggested here
https://github.com/mesa/mesa/issues/3806
